src/asm/parser.y

```diff
@@ -1026,15 +1026,15 @@ align_spec	: uconst {
 				$$.alignOfs = 0;
 			}
 		}
-		| uconst T_COMMA uconst {
+		| uconst T_COMMA const {
 			if ($1 > 16) {
 				error("Alignment must be between 0 and 16, not %u\n", $1);
-			} else if ($3 >= 1 << $1) {
-				error("Alignment offset (%u) must be smaller than alignment size (%u)\n",
-					$3, 1 << $1);
+			} else if ($3 <= -(1 << $1) || $3 >= 1 << $1) {
+				error("The absolute alignment offset (%d) must be less than alignment size (%u)\n",
+					$3 < 0 ? -$3 : $3, 1 << $1);
```

Comment:
I think there's something to do with `INT_MIN` here... how about converting the result to `uintN_t` and printing using `%" PRIuN`?

Reply:
I want a `DWIM` function/macro/language extension for numbers.